components/raftstore/src/store/fsm/peer.rs

```diff
@@ -952,36 +953,46 @@ where
             }
         };
         let peer_id = self.fsm.peer.peer_id();
-        let f = self
-            .ctx
-            .timer
-            .delay(timeout)
-            .map(move |_| {
-                fail_point!(
-                    "on_raft_log_gc_tick_1",
-                    peer_id == 1 && tick == PeerTicks::RAFT_LOG_GC,
-                    |_| unreachable!()
+        let cb = Box::new(move || {
+            fail_point!(
+                "on_raft_log_gc_tick_1",
+                peer_id == 1 && tick == PeerTicks::RAFT_LOG_GC,
+                |_| unreachable!()
+            );
+            // This can happen only when the peer is about to be destroyed
+            // or the node is shutting down. So it's OK to not to clean up
+            // registry.
+            if let Err(e) = mb.force_send(PeerMsg::Tick(tick)) {
+                debug!(
+                    "failed to schedule peer tick";
+                    "region_id" => region_id,
+                    "peer_id" => peer_id,
+                    "tick" => ?tick,
+                    "err" => %e,
                 );
-                // This can happen only when the peer is about to be destroyed
-                // or the node is shutting down. So it's OK to not to clean up
-                // registry.
-                if let Err(e) = mb.force_send(PeerMsg::Tick(tick)) {
-                    debug!(
-                        "failed to schedule peer tick";
-                        "region_id" => region_id,
-                        "peer_id" => peer_id,
-                        "tick" => ?tick,
-                        "err" => %e,
+            }
+        });
+        let idx = tick.bits() as usize;
+        let batch = &mut self.ctx.tick_batch[idx];
+        if batch.ticks.is_empty() || batch.wait_duration == timeout {
```

Comment:
Maybe we can batch all of the timer and combine the same timeout timer.

Reply:
But the timeout of most ticks  are different...